src/GramMatrix.cc

```diff
@@ -18,37 +19,36 @@
 #include <unistd.h>
 #include <vector>
 
-void rotateSym(dist_matrix::DistMatrix<DISTMATDTYPE>& mat,
-    const dist_matrix::DistMatrix<DISTMATDTYPE>& rotation_matrix,
-    dist_matrix::DistMatrix<DISTMATDTYPE>& work);
-
-GramMatrix::GramMatrix(const int ndim) : dim_(ndim)
+template <class T>
```

Comment:
I think it would be good to use something more descriptive than `T` for the template parameter. It's really hard to for someone new to the code to know what `T` represents. That's something I have had difficulties with in other part of the code.

Reply:
Good point. Just changed that.